torch/distributed/checkpoint/filesystem.py

```diff
@@ -126,9 +127,12 @@ def __init__(self, resolve_fun, stream=None, inflight_threshhold=1_000_000):
         self.current_items: collections.deque = collections.deque()
         self.idx = 0
         self.started = False
-        self.stream = stream or torch.cuda.current_stream()
-        if self.stream != torch.cuda.current_stream():
-            self.stream.wait_stream(torch.cuda.current_stream())
+        self.device_type = (getattr(stream, "device", torch.device("cuda"))).type
+        assert self.device_type in ["cuda", torch._C._get_privateuse1_backend_name()]
```

Comment:
I'm trying to make the "cpu" device have the same API here: https://github.com/pytorch/pytorch/pull/103172

Once that change is merged, we can remove all those asserts and conditional checks for stream device.
I'd love to move all the code we're adding in utils.py to a top level module that would even include protocols that each device type needs to provide.

Reply:
> 

yes, that change looks cool. It  will make it easy to extend new device, we can process the `cpu` device same as `cuda` and `privateuse1` backend. It is also very useful for other modules.  And I will unify the code once your change is merged. @kumpera

Reply:
And another idea, maybe we could to this APIs with the same func but with a device argument, like `torch.stream(device="cuda")` for `cuda`, and `torch.stream(device="cpu")` for `cpu`. And it can unify the APIs, but it will make a big change. I do not know which is better now. @kumpera